From 12fcca5a99c97886a75c383543c1be80d31abe10 Mon Sep 17 00:00:00 2001
From: Sandro <devel@penguinpee.nl>
Date: Thu, 25 Sep 2025 17:32:28 +0200
Subject: [PATCH] Adapt tests for change in numpy.count_zero()

As of NumPy 2.3.0 `numpy.count_zero` returns a NumPy scalar instead of a
Python int.
---
 test/test_array_function.py | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/test/test_array_function.py b/test/test_array_function.py
index 524f8ee..e539676 100644
--- a/test/test_array_function.py
+++ b/test/test_array_function.py
@@ -4,6 +4,7 @@
 import numpy
 from numpoly import polynomial
 import numpoly
+from packaging.version import Version
 
 X, Y = numpoly.variable(2)
 
@@ -357,10 +358,16 @@ def test_count_nonzero(func_interface):
     """Tests for numpoly.count_nonzero."""
     poly1 = polynomial([[0, Y], [X, 1]])
     poly2 = polynomial([[0, Y, X, 0, 0], [3, 0, 0, 2, 19]])
-    assert_equal(func_interface.count_nonzero(poly1), 3, type_=int)
     assert_equal(func_interface.count_nonzero(poly1, axis=0), [1, 2])
     assert_equal(func_interface.count_nonzero(poly2, axis=0), [1, 1, 1, 1, 1])
-    assert_equal(func_interface.count_nonzero(X), 1, type_=int)
+    # As of NumPy 2.3.0 `count_nonzero` returns a scalar instead of int
+    # https://github.com/jonathf/numpoly/issues/126
+    if Version(numpy.__version__) >= Version("2.3.0"):
+        assert_equal(func_interface.count_nonzero(poly1), 3, type_=numpy.intp)
+        assert_equal(func_interface.count_nonzero(X), 1, type_=numpy.intp)
+    else:
+        assert_equal(func_interface.count_nonzero(poly1), 3, type_=int)
+        assert_equal(func_interface.count_nonzero(X), 1, type_=int)
 
 
 def test_cumsum(interface):
